plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -0,0 +1,238 @@
+                {UUIDType.get()},
+        };
+    }
+
+    private void assertBucketAndHashEquals(String icebergTypeName, Object value, Integer expectedMaxIntBucket)
+    {
+        Type icebergType = Types.fromPrimitiveString(icebergTypeName);
+        if (value != null && icebergType.typeId() == DECIMAL) {
+            value = new BigDecimal((String) value).setScale(((DecimalType) icebergType).scale());
+        }
+
+        assertBucketEquals(icebergType, value);
+        assertMaxIntBucketEquals(icebergType, value, expectedMaxIntBucket);
+    }
+
+    private void assertBucketEquals(Type icebergType, Object value)
+    {
+        assertBucketNumberEquals(icebergType, value, Integer.MAX_VALUE);
+
+        assertBucketNumberEquals(icebergType, value, 2);
+        assertBucketNumberEquals(icebergType, value, 7);
+        assertBucketNumberEquals(icebergType, value, 31);
+        assertBucketNumberEquals(icebergType, value, 32);
+        assertBucketNumberEquals(icebergType, value, 100);
+        assertBucketNumberEquals(icebergType, value, 10000);
+        assertBucketNumberEquals(icebergType, value, 524287); // prime number
+        assertBucketNumberEquals(icebergType, value, 1 << 30);
+    }
+
+    private void assertBucketNumberEquals(Type icebergType, Object value, int bucketCount)
+    {
+        Integer icebergBucket = computeIcebergBucket(icebergType, value, bucketCount);
+        Integer trinoBucket = computeTrinoBucket(icebergType, value, bucketCount);
+
+        assertEquals(
+                icebergBucket,
+                trinoBucket,
+                format("icebergType=%s, icebergBucket=%d, trinoBucket=%d", icebergType, icebergBucket, trinoBucket));
+    }
+
+    private void assertMaxIntBucketEquals(Type icebergType, Object value, Integer expectedMaxIntBucket)
+    {
+        Integer icebergBucketHash = computeIcebergBucket(icebergType, value, Integer.MAX_VALUE);
+        Integer trinoBucketHash = computeTrinoBucket(icebergType, value, Integer.MAX_VALUE);
+
+        // Ensure hash is stable and does not change
+        assertEquals(
+                icebergBucketHash,
+                expectedMaxIntBucket,
+                format("expected Iceberg %s(%s) bucket with %sd buckets to be %d, got %d", icebergType, value, Integer.MAX_VALUE, expectedMaxIntBucket, icebergBucketHash));
+
+        // Ensure hash is stable and does not change
+        assertEquals(
+                trinoBucketHash,
+                expectedMaxIntBucket,
+                format("expected Trino %s(%s) bucket with %sd buckets to be %d, got %d", icebergType, value, Integer.MAX_VALUE, expectedMaxIntBucket, trinoBucketHash));
+    }
+
+    private Integer computeIcebergBucket(Type type, Object value, int bucketCount)
+    {
+        Transform<Object, Integer> bucketTransform = Transforms.bucket(type, bucketCount);
+        assertTrue(bucketTransform.canTransform(type), format("bucket function %s is not able to transform type %s", bucketTransform, type));
```

Comment:
If you use AssertJ, note that the description describes the expectation (positive case), not the failure.

```java
assertThat(bucketTransform.canTransform(type))
        .as("bucket function %s can transform to type %s", bucketTransform, type)
        .isTrue();
```

You can also use
```java
assertThat(bucketTransform).matches(t -> t.canTransform(type))
```
but I feel like getting the message/description right would be harder than the original.